app/controllers/CategorisationPreparationController.scala

```diff
@@ -76,8 +76,9 @@ class CategorisationPreparationController @Inject() (
         shorterCategorisationInfo <-
           Future.fromTry(Try(request.userAnswers.get(CategorisationDetailsQuery(recordId)).get))
 
-        longerComCode            <- Future.fromTry(Try(request.userAnswers.get(LongerCommodityQuery(recordId)).get))
-        longerCategorisationInfo <-
+        oldLongerCategorisationInfoOpt = request.userAnswers.get(LongerCategorisationDetailsQuery(recordId))
+        longerComCode                 <- Future.fromTry(Try(request.userAnswers.get(LongerCommodityQuery(recordId)).get))
```

Comment:
Assuming this is a '<-' future from try for a reason, instead of just '='

Reply:
LongerCommodityQuery must exist as its set on the LongerCommodityCode HasCorrectGoods page, which is the step before this. If it doesn't exist, we can't process things here (well we could but would be a lot more hassle)

Reply:
Yes, Chris told me if `request.userAnswers.get(LongerCommodityQuery(recordId)).get` fails `Try` will handle it safely

Reply:
I didn't see Chris's reply :D

Reply:
Cool :)